src/main/scala/mesosphere/util/state/MesosLeaderInfo.scala

```diff
@@ -0,0 +1,30 @@
+package mesosphere.util.state
+
+/**
+  * Utility class for keeping track of a Mesos Leader
+  */
+trait MesosLeaderInfo {
+
+  def currentLeaderUrl: Option[String]
+
+  def load(host: String, port: Int): Unit
```

Comment:
`MasterInfo` has no information about protocol, that's why I decided to use host and port.